cmd/minishift/cmd/start.go

```diff
@@ -544,11 +544,14 @@ func registerHost(libMachineClient *libmachine.Client) {
 		log.Debug("Skipping registration due to enabled --skip-registration flag")
 		return
 	}
-	if err := cluster.Register(libMachineClient); err != nil {
+	supportRegistration, err := cluster.Register(libMachineClient)
+	if err != nil {
 		atexit.ExitWithMessage(1, fmt.Sprintf("Error to register VM: %v", err))
-	} else {
+	}
+	if supportRegistration {
 		minishiftConfig.InstanceConfig.IsRegistered = true
 		minishiftConfig.InstanceConfig.Write()
+		return
```

Comment:
> we shouldn't because cluster.go is part of minikube pkg and moving forward we are thinking to consume it as library so moving this code doesn't make sense.

In fact I am wondering what Register and Unregister to in cluster.go at all. There is no driving force here and they could be as easily be placed in the Minishift registration package. That would make a better separation of concerns and also reduce one more pain point thinking of splitting out all Minikube related. 

> Is there any specific reason we can not move the below code to cluster.go i.e. Register function

There is a good reason imo. The commands are in a certain way our orchestrators. The fact that we store whether a machine is registered is really orthogonal to the actual act of registration and un-registrating it. If you move the config handling into the Register methods, you now do to things in one methods. Makes it harder to test and also to re-use.

registration.RegisterHostVM